src/middlewares/cors.js

```diff
@@ -7,15 +7,16 @@ const defaults = {
 
 const getOrigin = (options, handler) => {
   handler.event.headers = handler.event.headers || {}
+  const originHeader = handler.event.headers.Origin || handler.event.headers.origin
   if (options.origins && options.origins.length > 0) {
-    if (handler.event.headers.hasOwnProperty('Origin') && options.origins.includes(handler.event.headers.Origin)) {
-      return handler.event.headers.Origin
+    if (originHeader && options.origins.includes(originHeader)) {
+      return originHeader
     } else {
       return options.origins[0]
     }
   } else {
-    if (handler.event.headers.hasOwnProperty('Origin') && options.credentials && options.origin === '*') {
-      return handler.event.headers.Origin
+    if (originHeader && options.credentials && options.origin === '*') {
+      return originHeader
```

Comment:
Please note, I haven't changed the original behavior, but this is a security vulnerability. Asterisk origin should never work with credentials! I can see why it seemed convenient, but it the library should gracefully fail instead of hacking the standard CORS behaviour.